src/main/fc/fc_rc.c

```diff
@@ -277,25 +277,113 @@ int calcRcSmoothingCutoff(float avgRxFrameTime, bool pt1)
     }
 }
 
+FAST_CODE bool rcSmoothingRxRateValid(int currentRxRefreshRate)
+{
+    return (currentRxRefreshRate >= RC_SMOOTHING_RX_RATE_MIN_US && currentRxRefreshRate <= RC_SMOOTHING_RX_RATE_MAX_US);
+}
+
+FAST_CODE_NOINLINE void rcSmoothingSetFilterCutoffs(rcSmoothingFilter_t *rcSmoothingData)
+{
+    const float dT = targetPidLooptime * 1e-6f;
+    uint16_t oldCutoff = rcSmoothingData->inputCutoffFrequency;
+    if (rxConfig()->rc_smoothing_input_cutoff == 0) {
+        rcSmoothingData->inputCutoffFrequency = calcRcSmoothingCutoff(rcSmoothingData->averageFrameTimeUs, (rxConfig()->rc_smoothing_input_type == RC_SMOOTHING_INPUT_PT1));
+    }
+    if ((rcSmoothingData->inputCutoffFrequency != oldCutoff) || !rcSmoothingData->filterInitialized) {
+        for (int i = 0; i < PRIMARY_CHANNEL_COUNT; i++) {
+            if ((1 << i) & interpolationChannels) {
+                switch (rxConfig()->rc_smoothing_input_type) {
+                    case RC_SMOOTHING_INPUT_PT1:
+                        if (!rcSmoothingData->filterInitialized) {
+                            pt1FilterInit(&rcSmoothingData->filterPt1[i], pt1FilterGain(rcSmoothingData->inputCutoffFrequency, dT));
+                        } else {
+                            pt1FilterUpdateCutoff(&rcSmoothingData->filterPt1[i], pt1FilterGain(rcSmoothingData->inputCutoffFrequency, dT));
+                        }
+                        break;
+                    case RC_SMOOTHING_INPUT_BIQUAD:
+                    default:
+                        if (!rcSmoothingData->filterInitialized) {
+                            biquadFilterInitLPF(&rcSmoothingData->filterBiquad[i], rcSmoothingData->inputCutoffFrequency, targetPidLooptime);
+                        } else {
+                            biquadFilterUpdateLPF(&rcSmoothingData->filterBiquad[i], rcSmoothingData->inputCutoffFrequency, targetPidLooptime);
+                        }
+                        break;
+                }
+            }
+        }
+    }
+    oldCutoff = rcSmoothingData->derivativeCutoffFrequency;
+    if ((rxConfig()->rc_smoothing_derivative_cutoff == 0) && (rxConfig()->rc_smoothing_derivative_type != RC_SMOOTHING_DERIVATIVE_OFF)) {
+        rcSmoothingData->derivativeCutoffFrequency = calcRcSmoothingCutoff(rcSmoothingData->averageFrameTimeUs, (rxConfig()->rc_smoothing_derivative_type == RC_SMOOTHING_DERIVATIVE_PT1));
+    }
+    if ((rcSmoothingData->derivativeCutoffFrequency != oldCutoff) || !rcSmoothingData->filterInitialized) {
```

Comment:
This nested if can be replaced by:
```cpp
if (!rcSmoothingData->filterInitialized) {
  // initialize
} else if (rcSmoothingData->derivativeCutoffFrequency != oldCutoff) {
  // update
}
```
Same for non-derivative case.

Reply:
For the derivative case it makes sense.  I'm not sure about the input filter case.  To change that around I'd need to have a `switch` statement in each part of the `if !initialized` vs. an `if` in each `case`.  Either that or shift it up one level and have an entire !initialized vs. update `if` and replicate the `for` loop inside of each.

Reply:
Ok, I assumed they were similar